core/src/main/java/org/testcontainers/containers/wait/HttpWaitStrategyImpl.java

```diff
@@ -0,0 +1,115 @@
+    private int statusCode;
+    private boolean tlsEnabled;
+
+    /**
+     * Waits for status code 200.
+     *
+     * @param path the path to check
+     * @return this
+     */
+    public HttpWaitStrategyImpl waitForOk(String path) {
+        return waitForStatusCode(path, HttpURLConnection.HTTP_OK);
+    }
+
+    /**
+     * Waits for the given status code.
+     *
+     * @param path       the path to check
+     * @param statusCode the expected status code
+     * @return this
+     */
+    public HttpWaitStrategyImpl waitForStatusCode(String path, int statusCode) {
+        this.path = path;
+        this.statusCode = statusCode;
+        return this;
+    }
+
+    /**
+     * Indicates that the status check should use HTTPS.
+     */
+    public HttpWaitStrategyImpl usingTls() {
+        this.tlsEnabled = true;
+        return this;
+    }
+
+    @Override
+    public void waitUntilReady() {
+        final Integer livenessCheckPort = getLivenessCheckPort();
+        if (null == livenessCheckPort) {
+            logger().warn("No exposed ports or mapped ports - cannot wait for status");
+            return;
+        }
+
+        final String url = buildLivenessUri(livenessCheckPort).toString();
+        logger().info("Waiting for {} seconds for URL: {}", startupTimeout, url);
+
+        // try to connect to the URL
+        try {
+            retryUntilSuccess((int) startupTimeout.getSeconds(), TimeUnit.SECONDS, () -> {
+                getRateLimiter().doWhenReady(() -> {
+                    try {
+                        final HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+
+                        connection.setRequestMethod("GET");
+                        connection.connect();
+
+                        if (statusCode != connection.getResponseCode()) {
+                            throw new RuntimeException(String.format("HTTP response code was: %s",
+                                    connection.getResponseCode()));
+                        }
+
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+                return true;
+            });
+
+        } catch (TimeoutException e) {
+            throw new ContainerLaunchException(String.format(
+                    "Timed out waiting for URL to be accessible (%s should return HTTP %s)", url, statusCode));
+        }
+    }
+
+    /**
+     * Build the URI on which to check if the container is ready.
+     *
+     * @param livenessCheckPort the liveness port
+     * @return the liveness URI
+     */
+    private URI buildLivenessUri(int livenessCheckPort) {
+        final String scheme = (tlsEnabled ? "https" : "http") + "://";
+        final String host = DockerClientFactory.instance().dockerHostIpAddress();
+
+        final String port;
+        if ((tlsEnabled && 443 == livenessCheckPort) || (!tlsEnabled && 80 == livenessCheckPort)) {
```

Comment:
Can't livenessCheckPort actually take any value, if it's not using the fixed host port container and actually randomly allocated?